### DISCUSSION TITLE
How can I format F# files using Ionide?

### DISCUSSION THREAD

**Discussion by lulunac27a** [ionide/ionide-vscode-fsharp#2050] [Q&A]
I'm looking to format F# files using Ionide, is there is a formatter for Ionide?

**Comment by baronfel** [comment]
Yes - Ionide will guide you to installing [Fantomas](https://github.com/fsprojects/fantomas), the widely-used formatter for the F# ecosystem. Once installed, Ionide will transparently use Fantomas to format any F# code anytime the editor triggers formatting for F# source files.

**Comment by lulunac27a** [comment]
There is a [Stack Overflow](https://stackoverflow.com/a/71753291) answer that uses `Ionide.Ionide-fsharp` in `editor.defaultFormatter` setting in VSCode.
